lib/internal/fs/promises.js

```diff
@@ -632,10 +635,13 @@ async function writeFile(path, data, options) {
   }
 
   if (path instanceof FileHandle)
-    return writeFileHandle(path, data);
+    return writeFileHandle(path, data, options.signal);
 
   const fd = await open(path, flag, options.mode);
-  return writeFileHandle(fd, data).finally(fd.close);
+  if (options.signal && options.signal.aborted) {
```

Comment:
```suggestion
  if (options.signal?.aborted) {
```

Reply:
Bacause no backporting and 12.x?

Reply:
`AbortController` is not available on v12, is it?